iOS dependencies

Hey!

Awesome project! I'm trying to run the iOS app (haven't tried Android yet) and I'm having some issues. Is there anything not in the README that's needed?

I'm running on stable.

When I run the iOS app in XCode I get

```
+ RUSTFLAGS='-Z embed-bitcode'
+ cargo +ios-arm64 build --target aarch64-apple-ios --release --lib
error: no such subcommand: `+ios-arm64`
+ cargo build --target=x86_64-apple-ios --release
   Compiling log v0.4.8
   Compiling libc v0.2.60
   Compiling core-foundation-sys v0.6.2
   Compiling cfg-if v0.1.9
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-ios` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit code: 1
```

I have installed `x86_64-apple-ios` and `aarch64-apple-ios`.

Hey!

Sorry for not getting back to you sooner but the issue I got seems like it should be something I could fix but I haven't solved it yet.

This is my output (All I have done is installed rust-bitcode and changed
```
-RUSTFLAGS="-Z embed-bitcode" cargo +ios-arm64 build --target aarch64-apple-ios --release --lib
+RUSTFLAGS="-Z embed-bitcode" cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib
```

Output

```
+ RUSTFLAGS='-Z embed-bitcode'
+ cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib
error: no such subcommand: `+ios-arm64-1.46.0`
+ cargo build --target=x86_64-apple-ios --release
   Compiling libc v0.2.60
   Compiling log v0.4.8
   Compiling core-foundation-sys v0.6.2
   Compiling cfg-if v0.1.9
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-ios` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if`
```

If I only run `cargo build --target=x86_64-apple-ios --release` in the `ios_app`-folder it compiles fine.

Any suggestions?

Using the latest instructions both `cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib` and `cargo build --target=x86_64-apple-ios --release` works in the terminal.

Maybe I'm doing something wrong earlier. I'll describe my exact flow.

```
$ git clone https://github.com/i-schuetz/rust_android_ios.git
$ cd rust_android_ios/ios_app
$ open ios_app.xcodeproj
```
Press `Build and Run` (a k a play)-button in XCode.

This is the output from XCode:

```
+ cd /Users/axel/rust_android_ios/ios_app/../
+ cargo build --target=x86_64-apple-ios --release
   Compiling core-foundation-sys v0.6.2
   Compiling libc v0.2.60
   Compiling log v0.4.8
   Compiling cfg-if v0.1.9
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-ios` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit code: 1
```

It has to be a permissions or `PATH`-error.
I have tried manually linking in the script, didn't work. I have manually set the permissions to 755 for the `ios-arm64-1.46.0`-toolchain folder. As you can see `rustup toolchain list` shows us the toolchain...

I added some things to the script but it didn't help.

```
+ cd /Users/axel/rust_android_ios/ios_app/../
+ rustup override set stable
info: using existing install for 'stable-x86_64-apple-darwin'
info: override toolchain for '/Users/axel/rust_android_ios' set to 'stable-x86_64-apple-darwin'

  stable-x86_64-apple-darwin unchanged - rustc 1.49.0 (e1884a8e3 2020-12-29)

+ rustup toolchain list
stable-x86_64-apple-darwin (default) (override)
nightly-x86_64-apple-darwin
ios-arm64-1.46.0
1.48.0-x86_64-apple-darwin
+ cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib
error: no such subcommand: `+ios-arm64-1.46.0`
+ cargo build --target=x86_64-apple-ios --release
   Compiling libc v0.2.60
   Compiling log v0.4.8
   Compiling core-foundation-sys v0.6.2
   Compiling cfg-if v0.1.9
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-ios` target may not be installed
```

--- a/README.md
+++ b/README.md
@@ -106,11 +106,25 @@ rustup target add aarch64-linux-android armv7-linux-androideabi x86_64-linux-and
 
 ## iOS specific steps
 
-- Add targets
+- Install rust 1.46.0 with MacOSX binary support
 
 ```
-rustup target add x86_64-apple-ios aarch64-apple-ios
+wget https://github.com/getditto/rust-bitcode/releases/download/v1.46.0/rust-ios-arm64-1.46.0.zip
+unzip rust-ios-arm64-1.46.0.zip
+cd rust-ios-arm64-1.46.0
+./install.sh
 ```
+
+As the binaries are not signed, you'll have to convince MacOSX that it's safe to run them.
+One solution is to do the following:
+
+1. `cd rust_android_ios/ios_app`
+2. `cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib`
+3.a. if it fails because MacOSX doesn't trust the binary, go to 
+   `System Preferences -> Security & Privacy` and `Allow to run binary`
+    then go to 2.
+3.b. if it fails with another error, ask Google
+3.c. if it compiles, go on
 
 - Run the project in Xcode. This will build Rust, put the binaries in the correct place and start the app.
 

--- a/ios_app/build-rust-xcode.sh
+++ b/ios_app/build-rust-xcode.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # Ensure that `cargo` is in PATH, using the default location.
-export PATH="$PATH:${HOME}/.cargo/bin"
+. "$HOME/.cargo/env"
 
 set -x
 
@@ -9,7 +9,7 @@ set -x
 cd "${SRCROOT}/../"
 
 # Build binaries
-RUSTFLAGS="-Z embed-bitcode" cargo +ios-arm64 build --target aarch64-apple-ios --release --lib
+cargo +ios-arm64-1.46.0 build --target aarch64-apple-ios --release --lib
 cargo build --target=x86_64-apple-ios --release
 
 # Create fat binary

--- a/ios_app/ios_app.xcodeproj/project.pbxproj
+++ b/ios_app/ios_app.xcodeproj/project.pbxproj
@@ -206,7 +206,7 @@
 			isa = PBXProject;
 			attributes = {
 				LastSwiftUpdateCheck = 1030;
-				LastUpgradeCheck = 1030;
+				LastUpgradeCheck = 1250;
 				ORGANIZATIONNAME = com.schuetz;
 				TargetAttributes = {
 					84B0671E22F9B3FE006DDC18 = {
@@ -340,6 +340,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
@@ -401,6 +402,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
@@ -448,7 +450,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = "com.schuetz.ios-app";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_OBJC_BRIDGING_HEADER = "/Users/ischuetz/dev/rust/rust_android_ios/ios_app/Rust-Bridging-Header.h";
+				SWIFT_OBJC_BRIDGING_HEADER = "Rust-Bridging-Header.h";
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};
@@ -470,9 +472,10 @@
 					"$(PROJECT_DIR)",
 					"$(PROJECT_DIR)/core",
 				);
+				ONLY_ACTIVE_ARCH = YES;
 				PRODUCT_BUNDLE_IDENTIFIER = "com.schuetz.ios-app";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_OBJC_BRIDGING_HEADER = "/Users/ischuetz/dev/rust/rust_android_ios/ios_app/Rust-Bridging-Header.h";
+				SWIFT_OBJC_BRIDGING_HEADER = "Rust-Bridging-Header.h";
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};

--- a/ios_app/ios_app.xcodeproj/xcshareddata/xcschemes/ios_app.xcscheme
+++ b/ios_app/ios_app.xcodeproj/xcshareddata/xcschemes/ios_app.xcscheme
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1150"
+   LastUpgradeVersion = "1250"
    version = "1.3">
    <BuildAction
       parallelizeBuildables = "YES"

--- a/ios_app/ios_app.xcodeproj/xcshareddata/xcschemes/ios_appTests.xcscheme
+++ b/ios_app/ios_app.xcodeproj/xcshareddata/xcschemes/ios_appTests.xcscheme
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1150"
+   LastUpgradeVersion = "1250"
    version = "1.3">
    <BuildAction
       parallelizeBuildables = "YES"

